frontend/workflows/k8s/src/update-probe.tsx

```diff
@@ -0,0 +1,385 @@
+  );
+}
+
+const DeploymentDetails: React.FC<WizardChild> = () => {
+  const { onSubmit, onBack } = useWizardContext();
+  const deploymentData = useDataLayout("deploymentData");
+  const deployment = deploymentData.displayValue() as IClutch.k8s.v1.Deployment;
+  const update = (key: string, value: boolean) => {
+    deploymentData.updateData(key, value);
+  };
+
+  const currentDeploymentData = useDataLayout("currentDeploymentData");
+
+  const { containers } = deployment.deploymentSpec.template.spec;
+
+  const [containerName, setContainerName] = React.useState(containers[0].name);
+  const [probeType, setProbeType] = React.useState("livenessProbe");
+
+  const [containerIndex, setContainerIndex] = React.useState(0);
+
+  React.useEffect(() => {
+    // save the original values of deployment spec
+    if (deployment) {
+      currentDeploymentData.assign(deployment);
+    }
+  }, []);
+
+  const currentDeployment = findContainer({
+    deploymentSpec: deployment.deploymentSpec,
+    containerName,
+  });
+
+  const listedProbes = [];
+
+  if (Object.keys(currentDeployment.livenessProbe).length !== 0) listedProbes.push("livenessProbe");
+  if (Object.keys(currentDeployment.readinessProbe).length !== 0)
+    listedProbes.push("readinessProbe");
+
+  const containerBase = `deploymentSpec.template.spec.containers[${containerIndex}]`;
+
+  return (
+    <WizardStep error={deploymentData.error} isLoading={deploymentData.isLoading}>
+      <strong>Deployment Details</strong>
+      <MetadataTable
+        onUpdate={update}
+        data={[
+          { name: "Name", value: deployment.name },
+          { name: "Namespace", value: deployment.namespace },
+          {
+            name: "Container Name",
+            value: (
+              <Select
+                label="Container Name"
+                name="containerName"
+                onChange={value => {
+                  setContainerName(value);
+                  setContainerIndex(containers.findIndex(container => container.name === value));
+                  deploymentData.updateData("containerName", value);
+                }}
+                options={containers.map(container => {
+                  return { label: container.name };
+                })}
+              />
+            ),
+          },
+          {
+            name: "Probe Type",
+            value: (
+              <Select
+                label="Probe Type"
+                name="probeType"
+                onChange={value => {
+                  setProbeType(value);
+                }}
+                options={listedProbes.map(probe => {
+                  return { label: probe };
+                })}
+              />
+            ),
+          },
+          {
+            name: "Initial Delay Seconds",
+            value:
+              probeType === "livenessProbe"
+                ? currentDeployment.livenessProbe.initialDelaySeconds
+                : currentDeployment.readinessProbe.initialDelaySeconds,
```

Comment:
Should be able to clean all of these up a bit, since we already have the probeType declared we can just use it versus checking each time

```suggestion
            value: currentDeployment[probeType].initialDelaySeconds,
```

Reply:
Update probetype attributes